strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactMetadataServiceSnapshotsTest.java

```diff
@@ -36,45 +59,21 @@
         extends TestCaseWithMavenArtifactGenerationAndIndexing
 {
 
-    private static final String REPOSITORY_SNAPSHOTS = "amss-snapshots";
+    private static final String TIMESTAMPED_SNAPSHOT_VERSION_FORMAT = "yyyyMMdd.HHmmss";
 
-    private static final String[] CLASSIFIERS = { "javadoc",
-                                                  "sources",
-                                                  "source-release" };
+    private static final String REPOSITORY_SNAPSHOTS = "amss-snapshots";
 
     private static final String ARTIFACT_BASE_PATH_STRONGBOX_METADATA = "org/carlspring/strongbox/strongbox-metadata";
 
-
     @Inject
     private ArtifactMetadataService artifactMetadataService;
 
-    private SimpleDateFormat formatter = new SimpleDateFormat("yyyyMMdd.HHmmss");
-
-    private Calendar calendar = Calendar.getInstance();
-
-
-    @BeforeEach
-    public void initialize()
-            throws Exception
-    {
-        createRepository(STORAGE0, REPOSITORY_SNAPSHOTS, RepositoryPolicyEnum.SNAPSHOT.getPolicy(), false);
-    }
-
     @Test
-    public void testSnapshotMetadataRebuild()
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class, ArtifactManagementTestExecutionListener.class })
+    public void testSnapshotMetadataRebuild(@MavenSnapshotRepository Repository repository,
+                                            @MavenSnapshotArtifactWithClassifiers(id = "org.carlspring.strongbox:strongbox-metadata") List<Path> snapshotArtifacts)
```

Comment:
I'm not sure I get the idea behind this `List<Path> ...`. I don't understand... 
Would you mind explaining this sourcery to me? :-o

Reply:
it explained in #1165 under "To create multiple test artifact versions" section.

We have `@MavenSnapshotArtifactWithClassifiers` here which is shortcut for `@MavenArtifactWithClassifiers` with predefined 5 snapshot versions. So the `List<Path>` will contain 5 elements, each for every version.

Reply:
If you have only one version then `List` not required and single `Path` can be injected.